### ISSUE TITLE
Hific Colab codes are outdated

### ISSUE THREAD

**Issue by yifeipet** [tensorflow/compression#125]
Hello Dr. Balle,

Please check the Colab codes. Can you change the codes and other necessary Tensorflow Compression codes (such as hyperprior models) to fully support Tensorflow 2? New NVIDIA cards do not support old CUDNN libraries that are for Tensorflow 1.15.  That can be a big issue if more people buy new NVIDIA cards that cannot support Tensorflow version 1. They might give up Tensorflow Compression or even Tensorflow, and go completely to Pytorch. 

Yes, there are Pytorch codes for Hyperprior and Hific. But they might not be official codes so the results by these codes might be not accurate. 

Tensorflow version 2 doesn't support LPIPS loss. The pretrained neural network for LPIPS is not in Tensorflow 2. Can you also provide some codes for that? 

I understand you are busy. But hope you can solve the most issues. Thank you!

Sincerely,
Yifei

**Comment by fab-jul** [comment]
Hi Yifei,
thanks for your interest in HiFiC. I will look into this this week. could you provide some details? It is my understanding that HiFiC does in theory support TF2, but LPIPS does not? 
Fabian

**Comment by yifeipet** [comment]
> Hi Yifei, thanks for your interest in HiFiC. I will look into this this week. could you provide some details? It is my understanding that HiFiC does in theory support TF2, but LPIPS does not? Fabian

Can you check this Tensorflow HiFic link: [https://github.com/tensorflow/compression/tree/master/models/hific](url)? This is only for Tensorflow 1.3.

For LPIPS loss, Tensorflow 2 doesn't support it. It's only supported by Tensorflow 1. Please check this link: [https://github.com/alexlee-gk/lpips-tensorflow](url). 

Would you mind to update you HiFic codes to Tensorflow 2 codes? 

I trained HiFic model using the Pytorch codes: [https://github.com/Justin-Tan/high-fidelity-generative-compression](url). But I found the big gap between the bpp using entropy loss and the real bpp. I also tested the pretrained model provided by Justin using Kodak dataset on the Colab demo. But I still found a big gap between the theoretical bpp and the real bpp. 

Can you please check the issue? You might just use the pretrained model from the Github colab demo to check it. If you agree with the HiFic Pytorch codes [https://github.com/Justin-Tan/high-fidelity-generative-compression,](url) should I use the theoretical bpp or the real bpp for my comparison experiments?

**Comment by yifeipet** [comment]
> Hi Yifei, thanks for your interest in HiFiC. I will look into this this week. could you provide some details? It is my understanding that HiFiC does in theory support TF2, but LPIPS does not? Fabian

Hi Fabrian,
Any update for HiFic codes? Thank you!

Sincerely,
Yifei

**Comment by hific** [comment]
Hi, there are currently no plans to update the published code. Is it not possible to run it by pinning TF to the version [from the requirements](https://github.com/tensorflow/compression/blob/master/models/hific/requirements.txt)?

You should always use real bpp (that's what we use for the paper). I do not know why the code from justin has a big difference there.

**Comment by MahmoudAshraf97** [comment]
Hi @yifeipet 
I successfully ported the LPIPS code from this repo to TF2.x by simply changing the outdated function calls to their respective TF2 counterparts, it works perfectly and I can mail you the new codes to use in your projects

**Comment by yifeipet** [comment]
> Hi @yifeipet I successfully ported the LPIPS code from this repo to TF2.x by simply changing the outdated function calls to their respective TF2 counterparts, it works perfectly and I can mail you the new codes to use in your projects

Great. Thank you! I will try this method to port LPIPS to Tensorflow 2.

**Comment by apbyju** [comment]
> Hi @yifeipet
> I successfully ported the LPIPS code from this repo to TF2.x by simply changing the outdated function calls to their respective TF2 counterparts, it works perfectly and I can mail you the new codes to use in your projects

Can you please share with me the codes that works with TF2.x

**Comment by apbyju** [comment]
Hi @MahmoudAshraf97 .. For the compression project I am working on I have a gpu-tf1 compatibility issues. Can you please add the TF2 HiFiC codes to your repo.

**Comment by MahmoudAshraf97** [comment]
Hi @apbyju, Unfortunately I'm busy with other work right now so I cant rewrite hific in TF2 but check my @MahmoudAshraf97/AutoencoderCompression repo it has codes to run LPIPS with any TF2 project
